### PR TITLE
Dynamic

### PR DESCRIPTION
**Pull request by rayangendre**
I made the dynamic stock pages in which I connected the APIS and added all the connections that were needed

### CODE REVIEW

File: FrontEnd/myapp/src/ArticlesList.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  return (
+    <List divided style={{ maxWidth: 900, margin: "0 auto" }}>
+      {props.articles.map((article, index) => (
+        <ArticleItem article={article} key={article.title + index} />
+      ))}
+    </List>
+  );
+};
+
+export { ArticleList };
```
Comment (c1jb) [inline FrontEnd/myapp/src/ArticlesList.js:44]: A lot was added here and what you did makes a lot of sense. Smart additions

File: FrontEnd/myapp/src/Stock.js [hunk 2]
```diff
@@ -66,8 +78,19 @@ class Stock extends React.Component {
...
+          <Header as="h2" style={{ textAlign: "center", margin: 20 }}>
+            News about {this.state.ticker}
+          </Header>
+          {this.state.articles.length > 0 && (
+            <ArticleList articles={this.state.articles} />
+          )}
+          {this.state.apiError && (
+            <p>Could not fetch any articles. Please try again.</p>
+          )}
+        </Container>
```
Comment (c1jb) [inline FrontEnd/myapp/src/Stock.js:93]: Smart to make sure this is updated with the other page

File: FrontEnd/myapp/src/Table.js [hunk 3]
```diff
@@ -67,7 +68,7 @@ export function WatchlistTable(props) {
               sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
             >
               <TableCell component="th" scope="row">
-                {row.name}
+                <Link to={"/dynamic/" + row.name}>{row.name}</Link>
```
Comment (c1jb) [inline FrontEnd/myapp/src/Table.js:71]: Good change.

File: FrontEnd/myapp/src/Stock.js [hunk 4]
```diff
@@ -9,11 +13,20 @@ class Stock extends React.Component {
...
 
-  componentDidMount() {
+  async componentDidMount() {
     this.fetchStock();
+    try {
+      const response = await articles(this.state.ticker);
+      this.setState({ articles: response.articles });
+    } catch (error) {
+      this.setState({ apiError: "Could not find anything" });
+    }
```
Comment (tbryant-tbryant) [inline FrontEnd/myapp/src/Stock.js:29]: This is important for error checking, good change.

File: FrontEnd/myapp/src/Table.js (hunk 3 above)
Comment (tbryant-tbryant) [inline FrontEnd/myapp/src/Table.js:71]: Good idea to add the link to the table, will be a good feature.

**Review by tbryant-tbryant — APPROVED** [review]
Everything looks good, there are some large building blocks that allow for tons of improvements and new features. OK to merge.

File: FrontEnd/myapp/src/Stock.js (hunk 2 above)
Comment (rayangendre) [reply FrontEnd/myapp/src/Stock.js:93]: I believe it will be good to have them synced together as so

File: FrontEnd/myapp/src/Table.js (hunk 3 above)
Comment (rayangendre) [reply FrontEnd/myapp/src/Table.js:71]: Thank you

File: FrontEnd/myapp/src/Stock.js (hunk 4 above)
Comment (rayangendre) [reply FrontEnd/myapp/src/Stock.js:29]: I think we need to add more error checking elsewhere in the function

**Pull request merged**